tests/integration/test_image_filters.py

```diff
@@ -612,6 +612,37 @@ def test_rag_threshold_filter_on_rgb_image(pil_image, expected_image):
     )
 
 
+@pytest.mark.parametrize(
+    "pil_image, expected_image, masked",
+    (
+        (
+            RGB.DIAGNOSTIC_SLIDE_THUMB_RGB,
+            "mask-arrays/diagnostic-slide-thumb-rgb-rag-threshold-labels",
+            False,
+        ),
+        (
+            RGB.DIAGNOSTIC_SLIDE_THUMB_RGB,
+            "mask-arrays/diagnostic-slide-thumb-rgb-rag-threshold-maskedlabels",
+            True,
+        ),
+    ),
+)
+def test_rag_threshold_filter_return_labels(pil_image, expected_image, masked):
+    expected_value = load_expectation(expected_image, type_="npy")
+
+    if masked:
```

Comment:
Conditionals in unit tests are considered a bad practice. See here: https://testsmells.org/pages/testsmells.html

> Test methods need to be simple and execute all statements in the production method. Conditions within the test method will alter the behavior of the test and its expected output, and would lead to situations where the test fails to detect defects in the production method since test statements were not executed as a condition was not met. Furthermore, conditional code within a test method negatively impacts the ease of comprehension by developers.

To avoid using conditionals you can refactor your test using parametrization or by defining multiple test methods.

In this case you could use parametrization, create the mask beforehand and store it as a fixture in `tests/fixtures/mask-arrays` and then use it as we use it here: https://github.com/histolab/histolab/blob/0f0c18c2b43a5b993674323f6eda2b7b50f53a23/tests/integration/test_morphological_filters.py#L14
while for the second case the mask is simply `None`

Reply:
OK done.